server/routerlicious/packages/lambdas/src/utils/validateDocument.ts

```diff
@@ -0,0 +1,24 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { IDocument, IServiceConfiguration } from "@fluidframework/server-services-core";
+
+/**
+ * Whether a document exists and is not functionally deleted.
+ */
+export function isDocumentValid(document: IDocument): boolean {
+    return !!document && document.scheduledDeletionTime !== undefined;
```

Comment:
I'm probably getting confused here, but originally, we would say "Received attempt to connect to missing/deleted document" in this situation: `if (!document || document.scheduledDeletionTime) {`

I think it means that if the `scheduledDeletionTime` is set, then it is functionally deleted. But here, we seems to be doing the opposite; we say that the document is valid if `scheduledDeletionTime` is set. Was that intentional?

Reply:
You're totally right. Great catch. Thanks, Henrique!